pkg/helpers/cmd/errors.go

```diff
@@ -30,3 +35,80 @@ func CheckPodSecurityErr(err error) {
 
 	kcmdutil.CheckErr(err)
 }
+
+func CustomOIDCDerivedErrorChecker(discoveryClient discovery.DiscoveryInterface, what string) func(error) {
+	return func(err error) {
+		CheckGroupResourceNotRegisteredErr(err, discoveryClient, what, CustomOIDCReason, oauthv1.GroupVersion.Group, userv1.GroupVersion.Group)
+	}
+}
+
+const (
+	CustomOIDCReason = "This may occur if an external OIDC provider is configured."
+)
+
+// CheckGroupResourceNotRegisteredErr checks to see if originalError stems from the fact that
+// the GR in question is not registered on the server.
+func CheckGroupResourceNotRegisteredErr(originalError error, discoveryClient discovery.DiscoveryInterface, what, why string, groups ...string) {
+	if agg, ok := originalError.(utilerrors.Aggregate); ok && len(agg.Errors()) == 1 {
+		originalError = agg.Errors()[0]
+	}
+
+	if originalError == nil {
+		return
+	}
+
+	// only NotFound errors can stem from missing GRs, so short-circuit if we're
+	// dealing with any other error
+	if !apierrors.IsNotFound(originalError) {
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	status, ok := originalError.(apierrors.APIStatus)
+	if !ok && !errors.As(originalError, &status) {
```

Comment:
Shouldn't this be
```suggestion
	if !ok || !errors.As(originalError, &status) {
```
instead?

Reply:
I wonder this case is already covered in `IsNotFound` above?

Reply:
No, the logic as written is correct. If it's not cast-able *and* `errors.As` does not work, then we short-circuit and return.